background/style-manager.js

```diff
@@ -320,19 +376,33 @@ const styleManager = (() => {
     });
   }
 
-  function saveStyle(style) {
+  function beforeSave(style) {
     if (!style.name) {
       throw new Error('style name is empty');
     }
     if (style.id == null) {
       delete style.id;
     }
+    if (!style._id) {
+      style._id = uuid();
+    }
+    style._rev = Date.now();
     fixUsoMd5Issue(style);
+  }
+
+  function afterSave(style, newId) {
+    if (style.id == null) {
+      style.id = newId;
+    }
+    uuidIndex.set(style._id, style.id);
+    sync.put(style._id, style._rev);
+  }
+
+  function saveStyle(style) {
+    beforeSave(style);
     return db.exec('put', style)
       .then(event => {
-        if (style.id == null) {
-          style.id = event.target.result;
-        }
+        afterSave(style, event.target.result);
```

Comment:
This line is throwing JS errors in the master branch now. The `event` parameter is actually a number (the style id, I think), so `event.target.result` shows up as an error when I try to save a style

![](https://user-images.githubusercontent.com/136959/68986523-da530300-07e5-11ea-8a84-fb2ae5c88d72.png)

Reply:
The problem is solved if I modify it to `event.target && event.target.result`, but I'm not sure what was expected here.

Reply:
Are you using Chrome storage? It seems that the return value is wrong here:
https://github.com/openstyles/stylus/blob/2742cbe120270a3de865f078c5f4ee15a7785ab2/background/db.js#L172